src/Surfnet/StepupMiddleware/ManagementBundle/Controller/InstitutionConfigurationController.php

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+final class InstitutionConfigurationController extends Controller
+{
+    public function reconfigureAction(Request $request)
+    {
+        $this->denyAccessUnlessGranted(['ROLE_MANAGEMENT']);
+
+        $configuration = JsonHelper::decode($request->getContent());
+
+        $violations = $this->getValidator()->validate($configuration, new ValidReconfigureInstitutionsRequest());
+        if ($violations->count() > 0) {
+            throw BadCommandRequestException::withViolations('Invalid reconfigure institutions request', $violations);
+        }
+
+        if (empty($configuration)) {
+            $this->getLogger()->notice(sprintf('No institutions to reconfigure: empty configuration received'));
+
+            return new JsonResponse([
+                'status'       => 'OK',
+                'processed_by' =>  $request->server->get('SERVER_NAME') ?: $request->server->get('SERVER_ADDR'),
+                'applied_at'   => (new DateTime())->format(DateTime::ISO8601),
+            ]);
+        }
+
+        $commands = [];
+        foreach ($configuration as $institution => $options) {
+            $command                                  = new ReconfigureInstitutionConfigurationOptionsCommand();
+            $command->UUID                            = (string) Uuid::uuid4();
+            $command->institution                     = $institution;
+            $command->useRaLocationsOption            = $options['use_ra_locations'];
+            $command->showRaaContactInformationOption = $options['show_raa_contact_information'];
+
+            $commands[] = $command;
+        }
+
+        $this->getLogger()->notice(
+            sprintf('Executing %s reconfigure institution configuration options commands', count($commands))
+        );
+
+        $this->handleCommands($commands);
+
+        return new JsonResponse([
+            'status'       => 'OK',
+            'processed_by' =>  $request->server->get('SERVER_NAME') ?: $request->server->get('SERVER_ADDR'),
+            'applied_at'   => (new DateTime())->format(DateTime::ISO8601),
+        ]);
+    }
+
+    /**
+     * @param Command[] $commands
+     * @throws Exception
+     */
+    private function handleCommands(array $commands)
+    {
+        $pipeline         = $this->getPipeline();
+        $connectionHelper = $this->getConnectionHelper();
+
+        $connectionHelper->beginTransaction();
+
+        foreach ($commands as $command) {
+            try {
+                $pipeline->process($command);
+            } catch (ForbiddenException $e) {
+                $connectionHelper->rollBack();
+
+                throw new AccessDeniedHttpException(
+                    sprintf('Processing of command "%s" is forbidden for this client', $command),
+                    $e
+                );
+            } catch (Exception $exception) {
+                $connectionHelper->rollBack();
+
+                throw $exception;
```

Comment:
What would you want to convert it to? We're just intercepting for the rollback, not because we actually want to reason about the exception :smile: